https://gitlab.freedesktop.org/dbus/zbus/-/commit/35d1b8778754

$ kooha
 ERROR kooha::backend::screencast_portal > Failed to create a screencast call: zbus error: I/O error: No such file or directory (os error 2)

--- cargo-crates/zbus-2.0.0-beta.6/src/address.rs.orig	1973-11-29 21:33:09 UTC
+++ cargo-crates/zbus-2.0.0-beta.6/src/address.rs
@@ -35,8 +35,9 @@ impl Address {
         match env::var("DBUS_SESSION_BUS_ADDRESS") {
             Ok(val) => Self::from_str(&val),
             _ => {
-                let uid = Uid::current();
-                let path = format!("unix:path=/run/user/{}/bus", uid);
+                let runtime_dir = env::var("XDG_RUNTIME_DIR")
+                    .unwrap_or_else(|_| format!("/run/user/{}", Uid::current()));
+                let path = format!("unix:path={}/bus", runtime_dir);
 
                 Self::from_str(&path)
             }
